### PR TITLE
swaynag: more error checking and simplification

### PR DESCRIPTION
**Pull request by noocsharp**

### CODE REVIEW

File: swaynag/config.c [hunk 1]
```diff
@@ -416,31 +417,23 @@ int swaynag_load_config(char *path, struct swaynag *swaynag, list_t *types) {
...
+			type = swaynag_type_get(types, &line[1]);
 			if (!type) {
-				type = swaynag_type_new(name);
+				type = swaynag_type_new(&line[1]);
 				list_add(types, type);
 			}
-			free(name);
 		} else {
+			assert(line[nread - 1] == '\n');
+			line[nread - 1] = '\0';
```
Comment (emersion) [inline swaynag/config.c:433]: The last line may be missing a final newline

Comment (noocsharp) [reply swaynag/config.c]: fixed

File: swaynag/config.c [hunk 2]
```diff
@@ -416,31 +416,24 @@ int swaynag_load_config(char *path, struct swaynag *swaynag, list_t *types) {
...
 			if (!type) {
-				type = swaynag_type_new(name);
+				type = swaynag_type_new(&line[1]);
 				list_add(types, type);
 			}
-			free(name);
 		} else {
+			if (line[nread - 1] == '\n') {
+				line[nread - 1] = '\0';
+			}
```
Comment (emersion) [inline swaynag/config.c:433]: I'd be more comfortable moving this back outside of the `if`. The last line may not be `\n`-terminated, and the `line[nread - 2]` logic above requires to think twice.

**Comment by emersion** [comment]
Looks good apart from this.

**Review by emersion — APPROVED** [review]
Thanks!

**Pull request merged**